### PR TITLE
Fix streaming err raciness

### PR DESCRIPTION
**Pull request by KnifeyMoloko**
This PR introduces commits related to the issue https://github.com/scylladb/scylla-cluster-tests/issues/5506 and should fix the failures we've been seeing in relation to `DecommissionStreamingErr` and other `*StreamingErr` nemesis. The failures came about after a `ParallelObject` timed out. This `ParallelObject` is used to run 2 functions in parallel - a trigger function and a log watching function that should interrupt the process started by the trigger function. Timeouts have been caused by the fact that the trigger functions use `nodetool` remoter commands and since the process started by these commands gets interrupted, these functions did not return to the caller before the `ParallelObject`'s own timeout kicked in. 
This PR's commits address this and an additional potential source of racy behavior by: a) moving the log follower to be initiated before the trigger function is called and b) by introducing timeouts to the `nodetool` remoter calls.

Jenkins staging job with the nemesis in question: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/maciej/job/2022.1-10gb-gce/

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Review by fgelcer — APPROVED** [review]
LGTM, and i have a small comment/question:
shouldn't the `nodetool` commands timeout be a variable, set in a "centralised" in the call itself other than hard coded to each `nodetool` call?

**Comment by KnifeyMoloko** [comment]
> LGTM, and i have a small comment/question: shouldn't the `nodetool` commands timeout be a variable, set in a "centralised" in the call itself other than hard coded to each `nodetool` call?

Not really, at least in my opinion. The different `nodetool` commands can run for widely different amounts of time and event then same command can be expected to run much longer in some scenarios (more data used, heavier load), so I think it's better to use timeout as it is used here.

**Review by vponomaryov — APPROVED** [review]
LGTM

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -3260,21 +3259,25 @@ def decommission_post_action():
...
+            log_follower=log_follower,
             disrupt_func=self.reboot_node,
             disrupt_func_kwargs={"target_node": self.target_node, "hard": True, "verify_ssh": True},
             delay=0
         )
+
         ParallelObject(objects=[trigger, watcher], timeout=1200).call_objects()
         if new_node := decommission_post_action():
-            new_node.run_nodetool("rebuild")
+            new_node.run_nodetool("rebuild", timeout=300, retry=3)
```
Comment (fruch) [inline sdcm/nemesis.py:3278]: @fgelcer @vponomaryov, I think this change wasn't related, and isn't needed. i.e. assuming rebuild would always finish in 5min is quite off...

Comment (fgelcer) [reply sdcm/nemesis.py:3278]: how much do you think we should add here? wait forever isn't a good option neither

Comment (fruch) [reply sdcm/nemesis.py:3278]: waiting for ever is better option then 5min, that is failing on the release.
in general we shouldn't be setting such limits without some solid data out of previous runs and from multiple test case.

anyhow in this case, it wasn't even the intend fix/goal.

Comment (vponomaryov) [reply sdcm/nemesis.py:3278]: I agree with @fgelcer , before it was not set, and now it is explicit.
@fruch which value for timeout do you propose?

Comment (fruch) [reply sdcm/nemesis.py:3278]: I propose to set it back to no timeout:
https://github.com/scylladb/scylla-cluster-tests/pull/5808

Assuming a timeout here without doing the work of checking/validating it with prior data is pointless and causing confusion, like it is right now during the release, @roydahan was a about to disable this nemesis cause it's similar to a different issue with have with rebuild.

you think putting a timeout here is important enough, great put it on a task/issue and we'll assign someone to do the work, slipping such a thing in PR aiming to fix something else is something we should avoid (and we even backported it, adding more confusion on older releases as well)

Comment (roydahan) [reply sdcm/nemesis.py:3278]: Of course we need to revert it.
How did we approve it?
Rebuild will not finish in 5 minutes for most of our cases.

**Pull request merged**